enterprise/server/ociregistry/ociregistry.go

```diff
@@ -245,22 +256,49 @@ func (r *registry) handleBlobRequest(ctx context.Context, w http.ResponseWriter,
 		}()
 	}
 
+	casDigest := &repb.Digest{
+		Hash:      blobSHA256Digest,
+		SizeBytes: blobSize,
+	}
+	bsClient := r.env.GetByteStreamClient()
+	resourceName := digest.NewResourceName(
+		casDigest,
+		"",
+		rspb.CacheType_CAS,
+		repb.DigestFunction_SHA256,
+	)
+	fetchingFullLayer := offset == 0 && (limit == 0 || limit >= length)
+	if fetchingFullLayer {
+		err = cachetools.GetBlob(ctx, bsClient, resourceName, w)
+	} else {
+		err = cachetools.GetPartialBlob(ctx, bsClient, resourceName, w, offset, limit)
+	}
```

Comment:
looks like `GetPartialBlob` could be used for both cases? renaming it from `GetPartialBlob` => `GetBlobRange` might also make it more clear that it could be any range within the blob, even the full blob.